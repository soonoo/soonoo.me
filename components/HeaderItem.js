import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const HeaderItem = ({ to, children }) => (
  <NavLink to={to}>
    {children}
  </NavLink>
);

HeaderItem.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

export default HeaderItem;

