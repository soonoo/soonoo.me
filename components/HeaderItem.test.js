import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import HeaderItem from './HeaderItem';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <HeaderItem to='/href'>
          Hello world
        </HeaderItem>
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

