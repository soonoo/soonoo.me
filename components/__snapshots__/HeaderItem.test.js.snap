// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/href"
  onClick={[Function]}
>
  Hello world
</a>
`;
